pkg/util/loop/loop_linux.go

```diff
@@ -101,8 +102,18 @@ func (loop *Device) AttachFromFile(image *os.File, mode int, number *int) error
 		return fmt.Errorf("failed to set close-on-exec on loop device %s: %s", path, err.Error())
 	}
 
-	if _, _, err := syscall.Syscall(syscall.SYS_IOCTL, uintptr(loopFd), CmdSetStatus64, uintptr(unsafe.Pointer(loop.Info))); err != 0 {
-		return fmt.Errorf("failed to set loop flags on loop device: %s", syscall.Errno(err))
+	maxRetries := 5
+	for i := 0; i < maxRetries; i++ {
+		if _, _, err := syscall.Syscall(syscall.SYS_IOCTL, uintptr(loopFd), CmdSetStatus64, uintptr(unsafe.Pointer(loop.Info))); err != 0 {
+			if err == syscall.EAGAIN {
+				// loop_set_status() can temporarily fail with EAGAIN -> sleep and try again
+				// (cf. https://github.com/karelzak/util-linux/blob/dab1303287b7ebe30b57ccc78591070dad0a85ea/lib/loopdev.c#L1355)
+				time.Sleep(25*time.Millisecond)
```

Comment:
@muffgaga Looks like `losetup` set a 250ms : `xusleep(250000)` but whatever I tested within a loop with 25 ms pause while stressing the VM and it works fine

Reply:
Sorry again → I look at some [other part](https://github.com/karelzak/util-linux/blob/dab1303287b7ebe30b57ccc78591070dad0a85ea/lib/loopdev.c#L1327) of losetup where 25ms was used.

Reply:
Really thank you very much for tackling this issue, very much appreciated. And please add yourself in https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md :)

Reply:
@muffgaga Linter complains about formatting error, you need to execute `gofmt -w pkg/util/loop/loop_linux.go` to fix that